zerver/models.py

```diff
@@ -1938,7 +1939,7 @@ def profile_data(self) -> ProfileData:
                     "order": field_data["order"],
                     "value": value,
                     "rendered_value": rendered_value,
-                    "display_in_profile_summary": field_data["display_in_profile_summary"],
+                    "display_in_profile_summary": display_in_profile_summay,
```

Comment:
I don't think these hunks are necessary; all the contexts in which this function is used ignore the field. I'm not entirely sure, but I think it'd be clearer to skip this change.